test/parallel_api/numeric/numeric.ops/inclusive_scan_by_segment.pass.cpp

```diff
@@ -53,31 +128,94 @@ struct test_inclusive_scan_by_segment
         }
     }
 
-    template <typename Iterator1, typename Iterator2, typename Size>
+#ifdef DUMP_CHECK_RESULTS
+    template <typename Iterator, typename Size>
+    void display_param(const char* msg, Iterator it, Size n)
+    {
+        using DifferenceType = typename DifferenceTypeT<Iterator>::DifferenceType;
+
+        std::cout << msg;
+        for (DifferenceType i = 0; i < n; ++i)
+        {
+            if (i > 0)
+                std::cout << ", ";
+            std::cout << it[i];
+        }
+        std::cout << std::endl;
+    }
+#endif // DUMP_CHECK_RESULTS
+
+    template <typename Iterator1, typename Iterator2, typename Iterator3, typename Size>
     void
-    check_values(Iterator1 host_keys, Iterator2 val_res, Size n)
+    check_values(Iterator1 host_keys, Iterator2 host_vals, Iterator3 val_res, Size n)
     {
-        //T keys[n1] = { 1, 2, 3, 4, 1, 1, 2, 2, 3, 3, 4, 4, 1, 1, 1, ...};
-        //T vals[n1] = { 1, 1, 1, 1, 1, 2, 1, 2, 1, 2, 1, 2, 1, 2, 3, ...};
+        using ValT = typename ::std::decay<decltype(host_vals[0])>::type;
 
-        int segment_length = 1;
-        auto expected_segment_sum = 1;
-        auto current_key = host_keys[0];
-        auto current_sum = 0;
-        for (int i = 0; i != n; ++i)
+        return check_values(host_keys, host_vals, val_res, n, ::std::plus<ValT>());
+    }
+
+    template <typename Iterator1, typename Iterator2, typename Iterator3, typename Size, typename BinaryOperationCheck>
+    void
+    check_values(Iterator1 host_keys, Iterator2 host_vals, Iterator3 val_res, Size n, BinaryOperationCheck op)
+    {
+        // https://docs.oneapi.io/versions/latest/onedpl/extension_api.html
+        // keys:   [ 0, 0, 0, 1, 1, 1 ]
+        // values: [ 1, 2, 3, 4, 5, 6 ]
+        // result: [ 1, 1 + 2 = 3, 1 + 2 + 3 = 6, 4, 4 + 5 = 9, 4 + 5 + 6 = 15 ]
+
+#ifdef DUMP_CHECK_RESULTS
+        std::cout << "check_values(n = " << n << ") : " << std::endl;
+        display_param("keys:   ", host_keys, n);
+        display_param("values: ", host_vals, n);
+        display_param("result: ", val_res,   n);
+#endif // DUMP_CHECK_RESULTS
+
+        if (n < 1)
+            return;
+
+        if (n == 1)
+        {
+            EXPECT_TRUE(host_vals[0] == val_res[0], "wrong effect from exclusive_scan_by_segment");
+            return;
+        }
+
+        using ValT = typename ::std::decay<decltype(host_vals[0])>::type;
```

Comment:
I believe we don't need explicit `ValT `type at all.

Reply:
Fixed